ibr/binToJs.js

```diff
@@ -111,52 +16,10 @@ function onChooseFile() {
   if (file) {
     const fr = new FileReader();
     fr.onload = function(evt) {
-      const ibrData = InternalBuildingRepresentation.read(
-          new Pbf( evt.target.result ));
-      const scene = new THREE.Scene();
-      const camera = new THREE.PerspectiveCamera(
-          75, window.innerWidth / window.innerHeight, 0.1, 10000 );
-      const renderer = new THREE.WebGLRenderer();
-      renderer.setSize( window.innerWidth, window.innerHeight );
-      document.body.appendChild( renderer.domElement );
-      const controls = new OrbitControls( camera, renderer.domElement );
-      controls.target.set( 0, 0.5, 0 );
-      controls.update();
-      controls.enablePan = false;
-      controls.enableDamping = true;
-
-      // for datafiles that have top level name is ""
-      if (ibrData.name === '') {
-        ibrData.name = 'ibrData.name';
-      }
-      const li = document.createElement('li'); // list element container
-      document.getElementById('layerList').appendChild(li);
-      const rootSpan = createLabel('span', ibrData.name, li);
-      rootSpan.setAttribute('class', 'arrow');
-      li.appendChild(rootSpan);
-      const ul = document.createElement('ul');
-      ul.setAttribute('class', 'nested');
-      ul.setAttribute('id', ibrData.name);
-      li.appendChild(ul);
-      rootSpan.addEventListener('click', function() {
-        rootSpan.parentElement.querySelector('.nested').classList
-            .toggle('active');
-        rootSpan.classList.toggle('expanded-arrow');
-      });
-      extractSingleStructureData(ibrData, ibrData.name, scene);
-
-      camera.position.set( 0, 0, 7000 );
-      camera.lookAt( 0, 0, 0 );
-      animate();
-
-      /**
-       * Renders the scene.
-       */
-      function animate() {
-        requestAnimationFrame( animate );
-        controls.update();
-        renderer.render( scene, camera );
-      }
+      const bin = evt.target.result;
+      IBRSDK.renderTopIBRStructure( bin,
+          0, document.getElementById('mainCanvas') ); //top level, index 0
+      IBRSDK.createSidebar( bin, document.getElementById('layerList') );
```

Comment:
nice! this makes it a lot easier for third party developers to start using this package. two notes:

1) this encapsulates the rendering side, but we're still just passing in the binary file. This means it's very hard for developers to inspect and manipulate the data. I would suggest pulling the data into a wrapper class (e.g. ibrObject = IBRSDK.parse(bin);), then passing specific things to the render (e.g. IBRSDK.render(ibrObject.getLayer('A-WALL-E')). Plus, not including the binary parsing will make writing integration tests easier

2) Also note that `renderTopIBRStructure` implicitly forces the user to have an understanding of IBR structure, which we want to avoid. A broader and more generic method like IBRSDK.render() or renderLayers() is better

Reply:
1. I will add the wrapper class and getter functions in a new PR for refactoring the data class. This PR will be for refactoring the rendering functions.